pkg/util/priority/priority.go

```diff
@@ -77,9 +77,10 @@ func getDefaultPriorityClass(ctx context.Context, client client.Client) (*schedu
 	// we pick the one with the lowest priority value.
 	var defaultPC *schedulingv1.PriorityClass
```

Comment:
Maybe keep a copy, instead of a pointer.

Also returning a copy instead of a pointer should be fine.

Reply:
I'm not sure if I understand it correctly but you meant like the one below?

```
	var defaultPC schedulingv1.PriorityClass
	var found bool
	for _, pci := range pcs.Items {
		if pci.GlobalDefault {
			if !found || defaultPC.Value > pci.Value {
				found = true
				defaultPC = pci
			}
		}
	}

	if !found {
		return nil, nil
	}
	
	return &defaultPC, nil
```

Reply:
yes, but also just `return defaultPC, nil`.

What you have is fine too though.

Reply:
I see! It currently [checks if the default priority class is nil or not](https://github.com/kubernetes-sigs/kueue/blob/0acb185df73c7175ae3ed0038771ace80b6ff9b7/pkg/util/priority/priority.go#L63) so returning `schedulingv1.PriorityClass` from the `getDefaultPriorityClass` may complicate the check a little. Let's stick with returning `*schedulingv1.PriorityClass` then 🙂

Reply:
@alculquicondor I've updated it and squashed the commits. Would you review the PR again? 🙏